webapp/src/containers/Rhs/Rhs.container.tsx

```diff
@@ -1,59 +1,321 @@
+    const hasMoreLinkedChannels = useMemo<boolean>(() => (
+        (totalLinkedChannels.length - (paginationQueryParams.page * defaultPerPage) === defaultPerPage)
+    ), [totalLinkedChannels]);
+
+    // Show disconnect dialog component
+    const {DialogComponent, showDialog, hideDialog} = useDialog({
+        onSubmitHandler: disconnectUser,
+        onCloseHandler: () => {
+            hideDialog();
+        },
+    });
+
+    const {isRhsLoading} = getIsRhsLoading(state);
+    const {isOpen} = getSnackbarState(state);
+    const {data} = getApiState(pluginApiServiceConfigs.whitelistUser.apiServiceName);
     const {presentInWhitelist} = data as WhitelistUserResponse;
+    const {data: linkedChannels, isLoading} = getApiState(pluginApiServiceConfigs.getLinkedChannels.apiServiceName, getLinkedChannelsParams as SearchLinkedChannelParams);
+
+    // Handle searching of linked channels with debounce
+    useEffect(() => {
+        if (firstRender) {
+            return;
+        }
+
+        const timer = setTimeout(() => {
+            resetStates();
+        }, debounceFunctionTimeLimit);
+
+        /* eslint-disable consistent-return */
+        return () => {
+            clearTimeout(timer);
+        };
+    }, [searchLinkedChannelsText]);
+
+    // Make api call to get linked channels
+    useEffect(() => {
+        const linkedChannelsParams: SearchLinkedChannelParams = {page: paginationQueryParams.page, per_page: paginationQueryParams.per_page};
+        if (searchLinkedChannelsText) {
+            linkedChannelsParams.search = searchLinkedChannelsText;
+        }
 
-    // NOTE: Commented out on purpose.This is part of Phase-II
-    // const isAnyChannelLinked = useMemo(() => Boolean((linkedChannels as ChannelLinkData[])?.length), [linkedChannels]);
-    const isAnyChannelLinked = false;
+        setGetLinkedChannelsParams(linkedChannelsParams);
+        makeApiRequestWithCompletionStatus(pluginApiServiceConfigs.getLinkedChannels.apiServiceName, linkedChannelsParams);
+    }, [paginationQueryParams]);
 
+    // Update connected state and show alert on successful connection of the user
+    useEffect(() => {
+        if (connected && !isAlreadyConnected) {
+            showAlert({message: 'Your account is connected successfully.', severity: 'default'});
+            dispatch(setConnected({connected, msteamsUserId, username, isAlreadyConnected: true}));
+        }
+    }, [connected, isAlreadyConnected]);
+
+    // Update total linked channels after completion of the api to get linked channels
+    useApiRequestCompletionState({
+        serviceName: pluginApiServiceConfigs.getLinkedChannels.apiServiceName,
+        payload: getLinkedChannelsParams as SearchLinkedChannelParams,
+        handleSuccess: () => {
+            if (linkedChannels) {
+                setTotalLinkedChannels([...totalLinkedChannels, ...(linkedChannels as ChannelLinkData[])]);
+            }
+            if (firstRender && !paginationQueryParams.page) {
```

Comment:
for checking if RHS is opened for the first time or not we can make use of `view.rhs.isOpen` and `usePreviousState`

see if it works

Reply:
I think `views.rhs.open` will also get true when other plugins rhs also get opened. I don't think we can use it.